### ISSUE TITLE
name_usage ignores datasetKey and uuid - names from backbone returned

### ISSUE THREAD

**Issue by damianooldoni** [ropensci/rgbif#290]
After updating my fork from `upstream/master` (from 0.9.9.9221 to 0.9.9.9310) I noticed a change of the behavior of function `name_usage()`. A bug, I think.
For example, I would like to retrieve the data about 10 names from a dataset by `datasetKey`. I write the following line:
```r
test <- name_usage(datasetKey = "9ff7d317-609b-4c08-bd86-3bc404b77c42", limit = 10, return = "data")
```
Then, controlling the `datasetKey` column:
```r
unique(test$datasetKey)
```
I get the datasetKey of backbone: `d7dddbf4-2cf0-4f39-9b2a-bb099caae36c` instead of `9ff7d317-609b-4c08-bd86-3bc404b77c42`.
This is due to the fact that the URL in `temp <- cli$get(query = args)` (in function `gbif_GET()`) doesn't contain the word *datasetKey* before "=":
```
<crul response> 
  url: https://api.gbif.org/v1/species?limit=10&=9ff7d317-609b-4c08-bd86-3bc404b77c42
```
GBIF returns no errors, but points to its backbone by default in case of invalid parameters.

Now, I could solve the problem by adding the field `datasetKey` in the list passed to `rgbif_compact()`:
```r
args <- rgbif_compact(list(offset = start, limit = limit,
                             sourceId = sourceId,
                             datasetKey = datasetKey))
```
I find this behvaior quite strange because at first glance I don't see any changes in `name_usage()` and `getdata()` for the case I used, i.e. `data = "all"` (default) and `key = NULL`. 

Could you please give a look? Thanks a lot in advance! 

Session info follows.
<details> <summary><strong>Session Info</strong></summary>

```r
Session info ----------------------------------------------------------------
 setting  value                       
 version  R version 3.4.2 (2017-09-28)
 system   x86_64, mingw32             
 ui       RStudio (1.1.383)           
 language (EN)                        
 collate  Dutch_Belgium.1252          
 tz       Europe/Paris                
 date     2018-03-07                  

Packages --------------------------------------------------------------------
 package    * version    date       source                        
 assertthat   0.2.0      2017-04-11 CRAN (R 3.4.2)                
 base       * 3.4.2      2017-09-28 local                         
 bindr        0.1        2016-11-13 CRAN (R 3.4.2)                
 bindrcpp   * 0.2        2017-06-17 CRAN (R 3.4.2)                
 cli          1.0.0      2017-11-05 CRAN (R 3.4.2)                
 colorspace   1.3-2      2016-12-14 CRAN (R 3.4.2)                
 compiler     3.4.2      2017-09-28 local                         
 crayon       1.3.4      2017-09-16 CRAN (R 3.4.2)                
 crul         0.5.2      2018-02-24 CRAN (R 3.4.3)                
 curl         3.1        2017-12-12 CRAN (R 3.4.3)                
 data.table   1.10.4-3   2017-10-27 CRAN (R 3.4.2)                
 datasets   * 3.4.2      2017-09-28 local                         
 devtools     1.13.4     2017-11-09 CRAN (R 3.4.2)                
 digest       0.6.13     2017-12-14 CRAN (R 3.4.3)                
 dplyr      * 0.7.4      2017-09-28 CRAN (R 3.4.3)                
 drat         0.1.4      2017-12-16 CRAN (R 3.4.3)                
 fortunes     1.5-4      2016-12-29 CRAN (R 3.4.1)                
 geoaxe       0.1.0      2016-02-19 CRAN (R 3.4.2)                
 ggplot2      2.2.1      2016-12-30 CRAN (R 3.4.2)                
 git2r        0.19.0     2017-07-19 CRAN (R 3.4.2)                
 glue         1.2.0      2017-10-29 CRAN (R 3.4.2)                
 graphics   * 3.4.2      2017-09-28 local                         
 grDevices  * 3.4.2      2017-09-28 local                         
 grid         3.4.2      2017-09-28 local                         
 gtable       0.2.0      2016-02-26 CRAN (R 3.4.2)                
 httr         1.3.1      2017-08-20 CRAN (R 3.4.2)                
 jsonlite     1.5        2017-06-01 CRAN (R 3.4.2)                
 knitr        1.17       2017-08-10 CRAN (R 3.4.2)                
 lattice      0.20-35    2017-03-25 CRAN (R 3.4.2)                
 lazyeval     0.2.1      2017-10-29 CRAN (R 3.4.2)                
 magrittr     1.5        2014-11-22 CRAN (R 3.4.2)                
 memoise      1.1.0      2017-04-21 CRAN (R 3.4.2)                
 methods    * 3.4.2      2017-09-28 local                         
 munsell      0.4.3      2016-02-13 CRAN (R 3.4.2)                
 oai          0.2.2      2016-11-24 CRAN (R 3.4.2)                
 pillar       1.1.0      2018-01-14 CRAN (R 3.4.3)                
 pkgconfig    2.0.1      2017-03-21 CRAN (R 3.4.2)                
 plyr         1.8.4      2016-06-08 CRAN (R 3.4.2)                
 purrr        0.2.4      2017-10-18 CRAN (R 3.4.3)                
 R6           2.2.2      2017-06-17 CRAN (R 3.4.2)                
 Rcpp         0.12.14    2017-11-23 CRAN (R 3.4.3)                
 rgbif      * 0.9.9.9310 2018-03-07 local (ropensci/rgbif@ccfc7c8)
 rgeos        0.3-26     2017-10-31 CRAN (R 3.4.3)                
 rlang        0.1.6      2017-12-21 CRAN (R 3.4.3)                
 rstudioapi   0.7        2017-09-07 CRAN (R 3.4.2)                
 scales       0.5.0      2017-08-24 CRAN (R 3.4.3)                
 sp           1.2-6      2018-01-05 CRAN (R 3.4.3)                
 stats      * 3.4.2      2017-09-28 local                         
 stringi      1.1.6      2017-11-17 CRAN (R 3.4.2)                
 stringr      1.3.0      2018-02-19 CRAN (R 3.4.3)                
 tibble       1.4.2      2018-01-22 CRAN (R 3.4.3)                
 tools        3.4.2      2017-09-28 local                         
 triebeard    0.3.0      2016-08-04 CRAN (R 3.4.2)                
 urltools     1.6.0      2016-10-17 CRAN (R 3.4.2)                
 utf8         1.1.3      2018-01-03 CRAN (R 3.4.3)                
 utils      * 3.4.2      2017-09-28 local                         
 whisker      0.3-2      2013-04-28 CRAN (R 3.4.2)                
 withr        2.1.1      2017-12-19 CRAN (R 3.4.3)                
 xml2         1.2.0      2018-01-24 CRAN (R 3.4.3)                
 yaml         2.1.16     2017-12-12 CRAN (R 3.4.3)
```
</details>

**Comment by sckott** [comment]
Sorry about that , will have a look right away

**Comment by damianooldoni** [comment]
Problem tackled and solved in #291.
In general, I understood this from your code:
1. parameters with one or many values should be processed via `as_many_args()`
2. parameters with only one value should be processed via `rgbif_compact()`
And only at the end should be put together in `args <- c(args, ...)`

**Comment by sckott** [comment]
right, though we need to make sure that many repeated instances of the same argument are supported
